Sources/PT.PM.CSharpParseTreeUst/Converter/Helper.cs

```diff
@@ -35,6 +41,100 @@ protected TypeToken ConvertType(Ust node)
             return null;
         }
 
+        /// <summary>
+        /// When a name is absent, deconstruct a given tuple, otherwise create virtual member references
+        /// </summary>
+        private Ust HandleTupleStatement(VariableDeclarationSyntax node)
+        {
+            List<AssignmentExpression> variables =
+                node.Variables.Select(v => (AssignmentExpression)VisitAndReturnNullIfError(v)).ToList();
+
+            var typeNode = (TupleTypeSyntax)node.Type;
+            var textSpan = node.GetTextSpan();
+
+            for (int i = 0; i < typeNode.Elements.Count; i++)
+            {
+                TupleElementSyntax typeElement = typeNode.Elements[i];
+                if (typeElement.Identifier == null)
+                {
+                    continue;
+                }
+
+                CreateVirtualMemberRefs(ref variables, typeElement, i, textSpan);
+            }
+            var declaration = new VariableDeclarationExpression(null, variables, textSpan);
+            return new ExpressionStatement(declaration);
+        }
+
+        private void CreateVirtualMemberRefs(ref List<AssignmentExpression> variables, TupleElementSyntax typeElement, int initializerNumber, TextSpan textSpan)
+        {
+            variables.ForEach(variable =>
```

Comment:
Do not use linq methods with side effects, specifically `ForEach`. Use an ordinary `foreach` loop instead.